### PR TITLE
[test] Add invalid cases with br_table type mismatch

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: test/core/br_table.wast [hunk 1]
```diff
@@ -1552,6 +1562,35 @@
...
 )
 
+(assert_invalid
+  (module
+    (func (param i32) (result i32)
+      (loop (result i32)
+        (block (result i32)
+          i32.const 1
+          local.get 0
+          br_table 0 1
```
Comment (rossberg) [inline test/core/br_table.wast:1572]: Nit: use folded form here as rest of file?

File: test/core/br_table.wast [hunk 2]
```diff
@@ -1552,6 +1562,35 @@
...
+  "type mismatch"
+)
+(assert_invalid
+  (module
+    (func (param i32) (result i32)
+      (block (result i32)
+        (loop (result i32)
+          i32.const 1
+          local.get 0
+          br_table 0 1
```
Comment (rossberg) [inline test/core/br_table.wast:1586]: Same here.

File: test/core/br_table.wast (hunk 1 above)
Comment (gumb0) [reply test/core/br_table.wast]: Changed.

File: test/core/br_table.wast (hunk 2 above)
Comment (gumb0) [reply test/core/br_table.wast]: This is fixed, too.

**Review by rossberg — APPROVED** [review]
Thanks!

**Pull request merged**